pkg/controller/windowsmachineconfig/status.go

```diff
@@ -0,0 +1,86 @@
+package windowsmachineconfig
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"strings"
+
+	wmcapi "github.com/openshift/windows-machine-config-operator/pkg/apis/wmc/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// StatusManager is in charge of updating the WindowsMachineConfig object status
+type StatusManager struct {
+	// client is the client used to interact with the API
+	client client.Client
+	// wmcName is the name of the WMC resource we will update
+	wmcName types.NamespacedName
+	// joinedVMCount is the count of Windows VMs that have joined the cluster as a node
+	joinedVMCount int
+	// degradedCondition is the condition of type "degraded" to apply to the WMC object status
+	degradedCondition wmcapi.WindowsMachineConfigCondition
+	// conditionsToSet are all conditions except for the degraded condition to apply to the WMC object status
+	conditionsToSet []wmcapi.WindowsMachineConfigCondition
+}
+
+// NewStatusManager returns a new instance of StatusManager
+func NewStatusManager(client client.Client, name types.NamespacedName) *StatusManager {
+	return &StatusManager{
+		client:  client,
+		wmcName: name,
+	}
+}
+
+// setStatusConditions sets the status conditions applied when syncStatus() is called.
+func (s *StatusManager) setStatusConditions(conditions []wmcapi.WindowsMachineConfigCondition) {
+	if conditions != nil {
+		s.conditionsToSet = conditions
+	}
+}
+
+// syncStatus syncs the current status with the cluster and removes the reconciling status condition
```

Comment:
I don't follow how it removes the reconciling status condition. I also don't understand why you renamed it to `syncStatus()`.